src/base/PieTypes.hh

```diff
@@ -30,7 +30,16 @@ enum class Ownership
     const_reference, //!< Immutable reference to the data
 };
 
-//! Size/offset type for pie
+/*!
+ * Element indexing type for Pie access.
+ *
+ * The size type is plain "unsigned int" (32-bit in CUDA) rather than
+ * \c celeritas::size_type (64-bit) because CUDA currently uses native 32-bit
+ * pointer arithmetic. In general this should be the same type as the default
+ * OpaqueId::value_type. It's possible that in large problems 4 billion
+ * elements won't be enough (for e.g. cross sections), but in that case the
+ * PieBuilder will throw an assertion during construction.
```

Comment:
Is this comment referring to lines like:
```
void PieBuilder<T, M>::reserve(size_type count)
{
    CELER_EXPECT(count <= max_pie_size());
}
```
Shouldn't it be:
```
void PieBuilder<T, M>::reserve(size_type count)
{
    CELER_VALIDATE(count <= max_pie_size());
}
```
?  Or is the (assumingly, maybe indirectly via a data file) user provided `count` intercepted sooner?

Reply:
There's two layers of checking: `reserve` and `push_back` have debug-level assertions (they're going to get called often and don't need to provide detailed messages). As a second line of defense, for production runs where we might have larger objects or whatever, the `Pie` assignment operator does a `CELER_VALIDATE` check on the sizes. Both failure modes are tested in `Pie.test.cc`.